### DISCUSSION TITLE
Adding individualized HCP atlas

### DISCUSSION THREAD

**Discussion by yvalter** [simnibs/simnibs#364] [9. Results and analysis] [answered]
I want to analyze SimNIBS results with an individualized HCP atlas generated with Freesurfer and FSL. I have individualized annot files for each subject, but when I run the MATLAB analysis using my annot files, I receive an error message "ValueError: dimension mismatch"
  
```surf = mesh_load_gmsh4('C:\Users\yvalt\m2m_yishai\subject_overlays\yishai_TMS_1-0001_Magstim_70mm_Fig8_nii_scalar_central.msh')
[labels, snames] = subject_atlas(surf, 'C:\Users\yvalt\m2m_yishai', 'HCP_MMP1');
region_name = 'lh.46';
roi_idx=find(strcmpi(snames, region_name));
node_idx = labels.node_data{end}.data==roi_idx;
surf.node_data{end+1}.data = int8(node_idx);
surf.node_data{end}.name = region_name;
mesh_show_surface(surf, 'field_idx', region_name)
nodes_areas = mesh_get_node_areas(surf);
field_name = 'E_magn';
field_idx = get_field_idx(surf, field_name, 'node');
field = surf.node_data{field_idx}.data;
avg_field_roi = sum(field(node_idx).*nodes_areas(node_idx))/sum(nodes_areas(node_idx));
fprintf('Average %s in %s: %f\n', field_name, region_name, avg_field_roi)
reading binary
Number of Nodes              : 262594
Number of Triangles          : 525180
Number of Triangle Regions   : 1
Number of Tetrahedra         : 0
Number of Tetrahedron Regions: 0
Elapsed time is 0.106861 seconds.

surf = 

  struct with fields:

              node_data: {4×1 cell}
           element_data: {}
                  nodes: [262594×3 double]
              triangles: [525180×3 int32]
       triangle_regions: [525180×1 int32]
             tetrahedra: []
    tetrahedron_regions: []

Error using subject_atlas (line 39)
There was an error running subject_atlas:
 Traceback (most recent call last):
  File
  "C:\Users\yvalt\OneDrive\Desktop\SimNibs\simnibs_env\Lib\site-packages\simnibs\cli\subject_atlas.py",
  line 127, in <module>
    main()
  File
  "C:\Users\yvalt\OneDrive\Desktop\SimNibs\simnibs_env\Lib\site-packages\simnibs\cli\subject_atlas.py",
  line 115, in main
    labels_sub = morph.transform(labels).astype(int)
  File
  "C:\Users\yvalt\OneDrive\Desktop\SimNibs\simnibs_env\lib\site-packages\simnibs\utils\transformations.py",
  line 1557, in transform
    return self.morph_mat @ values
  File
  "C:\Users\yvalt\OneDrive\Desktop\SimNibs\simnibs_env\lib\site-packages\scipy\sparse\_base.py",
  line 630, in __matmul__
    return self._mul_dispatch(other)
  File
  "C:\Users\yvalt\OneDrive\Desktop\SimNibs\simnibs_env\lib\site-packages\scipy\sparse\_base.py",
  line 559, in _mul_dispatch
    raise ValueError('dimension mismatch')
ValueError: dimension mismatch```

**Comment by jduemose** [comment]
Hi, what are you trying to achieve using the `subject_atlas` function? This is a function for mapping an atlas from a reference space (fsaverage in this case) to subject space. Hence, if your atlas is already in subject space then this step is not needed (assuming you ran `charm` with the `--fs-dir` argument such that the freesurfer surfaces were used).

Best,
Jesper

**Comment by yvalter** [comment]
Thanks for the clarification. As a follow up question, how then can I use an individualized HCP atlas to extract a specific ROI for analysis and visualization? I have the rh and lh annot files as well as an STL file of the parcel of interest.

**Comment by jduemose** [comment]
Please have a look at [this](https://simnibs.github.io/simnibs/build/html/tutorial/analysis.html#roi-analysis-using-surfaces) tutorial. Given that your atlas is already in subject space, step 2.1 will be slightly different for you, i.e., instead of doing
```
[labels, snames] = subject_atlas(surf, fullfile(pwd, 'm2m_ernie'), 'HCP_MMP1');
```
you simply load your atlas directly, e.g., using `mesh_load_fsannot`.

**Comment by yvalter** [comment]
Thank you. I tried that and then received this error. I attached the mesh
annot files to this email for your reference.

surf =
mesh_load_gmsh4('C:\Users\yvalt\m2m_yishai\m2m_yishai_5\subject_overlays\yishai_TMS_1-0001_Magstim_70mm_Fig8_nii_scalar_central.msh')

atlas=
mesh_load_fsannot('C:\Users\yvalt\OneDrive\Desktop\SimNibs\simnibs_env\Lib\site-packages\simnibs\resources\
 yishai_TMS_1-0001_Magstim_70mm_Fig8_nii_scalar_central.msh
<https://drive.google.com/file/d/1Q5g2JDLbH58S9rnVnSRMFQ2QEn5HC_aA/view?usp=drive_web>
templates\fsaverage_atlases\lh.aparc_HCP_MMP1.annot')

Dot indexing is not supported for variables of this type.

Error in *mesh_load_fsannot* (line 62)
region_idx=unique(m.triangle_regions);

**Reply by jduemose** [reply, marked as answer]
You need to pass the (loaded) mesh to `mesh_load_fsannot` (your `surf` variable). From your code it seems that you pass the filename of the mesh, right?
